### ISSUE TITLE
Tests should not require root to run

### ISSUE THREAD

**Issue by ajor** [iovisor/bpftrace#127]

**Comment by danobi** [comment]
Do the tests require root? I'm having luck doing this:

```
./bpftrace_test --gtest_filter='-codegen*'
...
20 FAILED TESTS
```

The failed tests are another issue.

**Comment by ajor** [comment]
I think it's just codegen.builtin_name and codegen.builtin_name_wild which require root - they'll fail if you run them without it.

I've not looked into what they're doing, but I do see code which tries to read "/sys/kernel/debug/tracing/available_filter_functions". This should be mocked out like in the tracepoint_format_parser tests.

**Comment by danobi** [comment]
Sounds good, I'll give this a shot if that's ok.

**Comment by ajor** [comment]
Go for it!

**Comment by fbs** [comment]
fixed with b261833c014ad76c14a7d6c251ee37e38d89e77f
